Changes the way MPICC is set

dipietrantonio opened the pull request:
in newer pip versions `--install-option` does not exist, but the proposed method always works.

Commit:
Changes the way MPICC is set

in newer pip versions --install-option does not exist, but the proposed
method always works.

adamjstewart commented on var/spack/repos/builtin/packages/py-mpi4py/package.py:
```diff
@@ -31,17 +31,16 @@ class PyMpi4py(PythonPackage):
     depends_on("python@2.6:2.7,3.2:")
     depends_on("python@2.7:2.8,3.5:", when="@3.1:")
```
This can be removed, Spack only supports Python 3.7+

adamjstewart commented on var/spack/repos/builtin/packages/py-mpi4py/package.py:
```diff
@@ -31,17 +31,16 @@ class PyMpi4py(PythonPackage):
     depends_on("py-setuptools@40.9:", type="build")
-    # in newer pip versions --install-option does not exist
-    depends_on("py-pip@:23.0", type="build")
+    depends_on("py-pip", type="build")
```
This can be removed, the base class already adds pip

adamjstewart reviewed (commented):
@manuelakuhn FYI

Commit:
Removes outdated dependencies.

adamjstewart commented on var/spack/repos/builtin/packages/py-mpi4py/package.py:
```diff
@@ -28,20 +28,16 @@ class PyMpi4py(PythonPackage):
     # https://github.com/mpi4py/mpi4py/pull/311
     conflicts("^py-cython@3:")
 
     @when("@3.1:")
```
Let's remove this `@when` decorator. I added it in #28759 because the flag only worked for `python setup.py build`, not `pip install` in older versions. But the env var should actually work for all versions.

dipietrantonio replied:
Done.

Commit:
Removes `when` clause.

It was added in #28759 because the flag only worked for python setup.py build, not pip install in older versions. But the env var should actually work for all versions.
